provider/core/src/request.rs

```diff
@@ -28,13 +34,72 @@ impl fmt::Display for DataRequest<'_> {
     }
 }
 
+/// Metadata for data requests. This is currently empty, but it may be extended with options
+/// for tuning locale fallback, buffer layout, and so forth.
 #[derive(Default, Debug, Clone, Copy, PartialEq, Eq, PartialOrd, Ord)]
 #[non_exhaustive]
 pub struct DataRequestMetadata;
 
-/// A variant and language identifier, used for requesting data from a data provider.
+/// The main locale type used by the ICU4X data provider.
+///
+/// [`DataLocale`] contains less functionality than [`Locale`] but more than
+/// [`LanguageIdentifier`] for better size and performance while still meeting
+/// the needs of the ICU4X data pipeline.
+///
+/// # Examples
+///
+/// Convert a [`Locale`] to a [`DataLocale`] and back:
+///
+/// ```
+/// use icu_locid::Locale;
+/// use icu_provider::DataLocale;
+///
+/// let locale1 = "en-u-ca-buddhist".parse::<Locale>().unwrap();
+/// let data_locale: DataLocale = locale1.into();
+/// let locale2 = data_locale.into_locale();
+///
+/// assert_eq!(locale2.to_string(), "en-u-ca-buddhist");
+/// ```
+///
+/// You can alternatively create a [`DataLocale`] from a borrowed [`Locale`], which is more
+/// efficient than cloning the [`Locale`]:
+///
+/// ```
+/// use icu_locid::Locale;
+/// use icu_provider::DataLocale;
+///
+/// let locale1 = "en-u-ca-buddhist".parse::<Locale>().unwrap();
+/// let data_locale: DataLocale = (&locale1).into();
```

Comment:
Nit: I'd prefer writing this as
```rust
let data_locale = DataLocale::from(&locale);
```
which is a bit shorter and feels more explicit as well. `(&locale1).into()` requires the type information from outside the expression, which is cool if it's there, but not cool if you have to explicitly type the variable. Also in the other examples.

Reply:
+1, we should prefer from over `: DataLocale`

Reply:
The other two examples as well

Reply:
done